src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.UnrootedSymbolSet.cs

```diff
@@ -44,11 +45,15 @@ internal partial class SolutionState
             /// <summary>
             /// The <see cref="IAssemblySymbol"/>s or <see cref="IModuleSymbol"/>s produced through <see
             /// cref="Compilation.GetAssemblyOrModuleSymbol(MetadataReference)"/> for all the references exposed by <see
-            /// cref="Compilation.References"/>/
+            /// cref="Compilation.References"/>.  Sorted by the hash code produced by <see
+            /// cref="ReferenceEqualityComparer.GetHashCode(object?)"/> so that it can be binary searched efficiently.
             /// </summary>
-            public readonly WeakSet<ISymbol> SecondaryReferencedSymbols;
+            public readonly ImmutableArray<(int hashCode, WeakReference<ISymbol> symbol)> SecondaryReferencedSymbols;
```

Comment:
i did need to store as a tuple as i couldn't just have this be an array of weak refs.  the problem with the latter is that when binary searching, you might hit a weak ref that had been released, and there's no way for the binary search to know what to do there.

Reply:
The alternative is an `ImmutableArray<ReferenceHolder<ISymbol>>`

Reply:
i thought about that, but it required a lot of deeper plumpbing into that guy.  note: at this point, i think we could remove those types.